File: BOJ(Java 공부용)/1436.java
import java.io.*;

public class Main {

    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, -1, 0};
    static int[] goy = {1, 0, 0, -1};
    static int ans = 0;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int num = 666, cnt = 1;
        while (true) {
            if (cnt == n) break;
            num++;
            int temp = num;
            int six = 0;
            while (temp > 0) {
                int sk = temp % 10;
                if (sk == 6) six++;
                else if (six < 3) six = 0;
                temp /= 10;
            }
            if (six >= 3) cnt++;
        }
        bw.write(num + "");
        bw.flush();
        bw.close();
    }
}

File: BOJ(Python 공부용)/1436.py
#!/usr/bin/env python

n = input()
n = int(n)
num = 666
cnt = 1
while True:
    if cnt == n: break
    num+=1
    temp = num
    six = 0
    while temp!=0 :
        sk = temp%10
        if sk== 6: six+=1
        elif six<3 : six =0
        temp/=10
        temp = int(temp)
    if six >=3: cnt+=1

print("%d" %num)